aarch64: extend matmul heuristic checks to all neoverse platforms

snadampal opened the pull request:
for aarch64 neoverse platforms there are two gemm backends available
for matmul operator on PyTorch: (1) Arm Compute Library and (2) OpenBLAS.
While Arm Compute Library provides better performance over OpenBLAS,
it has overhead for the kernel launch time, and hence we use OpenBLAS
for smaller tensor compute. The heuristic was originally implemented for 
neoverse_v1. This commit extends the heuristic to other neoverse platforms

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

snadampal commented:
Hi @malfet , could you please review and approve if it looks good.

malfet commented:
@snadampal overall looks good, but can you please provide a short description on why this change is needed?

malfet commented on aten/src/ATen/native/LinearAlgebra.cpp:
```diff
@@ -1360,7 +1360,8 @@ static inline int64_t get_mkldnn_matmul_min_dim() {
 #if !defined(__s390x__)  && !defined(__powerpc__)
-      if (cpuinfo_initialize() && cpuinfo_get_uarchs_count() == 1 && cpuinfo_get_uarch(0)->uarch == cpuinfo_uarch_neoverse_v1) {
+      if (cpuinfo_initialize() && cpuinfo_get_uarchs_count() == 1 &&
+          (cpuinfo_get_uarch(0)->uarch == cpuinfo_uarch_neoverse_v1 || cpuinfo_get_uarch(0)->uarch == cpuinfo_uarch_neoverse_v2)) {
```
Can you please elaborate why this heuristic should be applied to neonvese_v2 but not to neonverse_n2?
Also, considering the code is almost identical between `get_mkldnn_matmul_min_dim` and `get_mkldnn_matmul_min_size` do you mind refactoring this into `is_arm_neonverse()` helper routine

malfet reviewed (changes requested):
@snadampal please ping me on PR when this change will apply to all neonverse arches and condition is refactored into a common function

snadampal committed:
aarch64: extend matmul heuristic checks to all neoverse platforms

for aarch64 neoverse platforms there are two gemm backends available
for matmul operator on PyTorch: (1) Arm Compute Library and (2) OpenBLAS.
While Arm Compute Library provides better performance over OpenBLAS,
it has overhead for the kernel launch time, and hence we use OpenBLAS
for smaller tensor compute. The heuristic was originally implemented for
neoverse_v1. This commit extends the heuristic to other neoverse platforms.

malfet reviewed (approved):
Thank you for the updates, looks good to me

malfet commented:
@pytorchbot merge

snadampal commented:
@malfet , thanks for initiating the merge. I was waiting for the CI runs to complete to ping you.
